pgrx/src/datum/array.rs

```diff
@@ -741,6 +741,13 @@ impl<T: IntoDatum + FromDatum> IntoDatum for Array<'_, T> {
     fn type_oid() -> Oid {
         T::array_type_oid()
     }
+
+    fn composite_type_oid(&self) -> Option<Oid> {
+        // the composite type oid for a vec of composite types is the array type of the base composite type
+        self.get(0)
+            .map(|v| v.composite_type_oid().map(|oid| unsafe { pg_sys::get_array_type(oid) }))
+            .flatten()
```

Comment:
It looks like it should work but doesn't this needlessly whiff for an Array of size zero when it could instead be
```suggestion
        Some(unsafe { pg_sys::get_array_type(self.raw.oid()) })
```

Reply:
hm.  Yeah, I suppose this would work for `Array`.  This was just a copy-paste from what I did for `Vec`, which in fact needs at least 1 value to figure this out.